Firestore/Swift/Tests/Integration/AsyncAwaitIntegrationTests.swift

```diff
@@ -120,5 +155,48 @@ let emptyBundle = """
       }
       XCTAssertThrowsError(try deleteAllIndexes(), "The client has already been terminated.")
     }
+
+    func testCanListenToDefaultSourceFirstAndThenCacheAsyncStream() async throws {
+      let collRef = collectionRef(withDocuments: [
+        "a": ["k": "a", "sort": 0],
+        "b": ["k": "b", "sort": 1],
+      ])
+
+      let query = collRef.whereField("sort", isGreaterThanOrEqualTo: 1).order(by: "sort")
+
+      // 1. Create a signal stream. The test will wait on this stream.
+      //    The Task will write to it after receiving the first snapshot.
+      let (signalStream, signalContinuation) = AsyncStream.makeStream(of: Void.self)
+
+      let stream = query.asyncThrowingStream()
+      var iterator = stream.makeAsyncIterator()
+
+      let task = Task {
```

Comment:
Is a Task here really needed?

I found doing it without Task and signalStream much easier to read:

https://github.com/firebase/firebase-ios-sdk/pull/14931/files#diff-c9933475816c94708983249c37c7ee449116ab4815fa96fb928113552f83c1eaR121

Reply:
No, the task is mainly used for testing the stream behaviour after the stream closed.